enclave/enclave-jaxrs/src/main/java/com/quorum/tessera/enclave/rest/RestfulEnclaveClient.java

```diff
@@ -173,10 +172,27 @@ public RawTransaction encryptRawPayload(byte[] message, PublicKey sender) {
         dto.setProvidedKey(recipientKey.getKeyBytes());
 
         final Response response = client.target(uri)
-            .path("addRecipient")
-            .request()
-            .post(Entity.json(dto));
+                .path("addRecipient")
+                .request()
+                .post(Entity.json(dto));
 
         return response.readEntity(byte[].class);
     }
+
+    @Override
+    public Status status() {
+        
+        final Response response = client.target(uri)
+                .path("ping")
+                .request().get();
+        
+        if(response.getStatus() == 200) {
```

Comment:
Not sure this is right. The `/ping` endpoint just returns the state of the remote Enclave. So if the Enclave has `STOPPED` status, the request would still be a `200`. Seems like the response needs to be evaluated back to the enum.

Reply:
We only use the http code to communicate state.